libsweep/Error.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <stdexcept>
+
+namespace sweep {
+struct ErrorBase : std::exception {
+  // parameter MUST be a STRING LITTERAL
+  template <size_t N> ErrorBase(const char (&msg)[N]) : _msg{&msg[0]} {}
+  const char* what() const noexcept override { return _msg; };
+
+private:
+  const char* _msg;
+};
```

Comment:
Yes, I pretty much copied the currrent error mechanism as I wasn't sure if the error messages being string litterals was a desired property or an undesired limitation of the current implementation.